lua/telescope/builtin/__lsp.lua

```diff
@@ -359,7 +359,14 @@ lsp.document_symbols = function(opts)
       return
     end
 
-    local locations = vim.lsp.util.symbols_to_items(result or {}, opts.bufnr) or {}
+    local locations = {}
+    if vim.fn.has "nvim-0.11" == 1 then
+      local offset_encoding = vim.lsp.get_clients({ bufnr = opts.bufnr })[1].offset_encoding
+      locations = vim.lsp.util.symbols_to_items(result or {}, opts.bufnr, offset_encoding) or {}
```

Comment:
```suggestion
      local client = assert(vim.lsp.get_client_by_id(ctx.client_id))
      locations = vim.lsp.util.symbols_to_items(result or {}, opts.bufnr, client.offset_encoding) or {}
```
Actually, we don't have to blindly just take the first client.
We can get the `client_id` from the `ctx` value that's passed into this handler callback:
```lua
vim.lsp.buf_request(opts.bufnr, "textDocument/documentSymbol", params, function(err, result, ctx, _)
```

We can do the same thing below as well

Reply:
Thanks for this - appears to be working now.